amt/clients/clients.py

```diff
@@ -52,3 +53,9 @@ async def get_task_by_urn(self, task_type: TaskType, urn: str, version: str = "l
             logger.exception(f"Invalid task {task_type.value} fetched: key 'urn' must occur in task {task_type.value}.")
             raise AMTInstrumentError()
         return response_data
+
+
+@alru_cache
+async def get_task_by_urn(task_type: TaskType, urn: str, version: str = "latest") -> dict[str, Any]:
+    client = TaskRegistryAPIClient()
```

Comment:
should we create a client for each call to this task or can we have a shared client?